### ISSUE TITLE
Remove IdentityServer from Blazor WASM app

### ISSUE THREAD

**Issue by jhulbertpmn** [abpframework/abp#10222]
* ABP Framework 4.4.3
* Blazor WASM
* EFCore (MS SQL) 

 I created a new Blazor WASM application, and followed the steps at https://community.abp.io/articles/replacing-identityserver4-with-identity-in-abp-arh0jxe1 to remove IdentityServer. The steps all appear to be successful, and the application starts up/runs. However, when I click login it takes me to https://localhost:44307/authentication/login-failed with a message: 

There was an error trying to log you in: ' (404)'

Is there an additional step that needs to be done specifically for Blazor WASM?

**Comment by rlavaud** [comment]
Blazor WASM needs some form of openidconnect to authenticate (identityserver, Azure AD B2C etc). The how-to you followed works for plain MVC app because ABP will fall back the plain old MS Identity.
From the Microsoft doc:
_Blazor WebAssembly supports authenticating and authorizing apps using OIDC via the Microsoft.AspNetCore.Components.WebAssembly.Authentication library._

[https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/?view=aspnetcore-5.0](url)

Long story short you need Identity Server with Blazor WASM

**Comment by jhulbertpmn** [comment]
> Blazor WASM needs some form of openidconnect to authenticate (identityserver, Azure AD B2C etc). The how-to you followed works for plain MVC app because ABP will fall back the plain old MS Identity. From the Microsoft doc: _Blazor WebAssembly supports authenticating and authorizing apps using OIDC via the Microsoft.AspNetCore.Components.WebAssembly.Authentication library._
> 
> [https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/?view=aspnetcore-5.0](url)
> 
> Long story short you need Identity Server with Blazor WASM

Makes sense! Ultimately I would like to use Azure AD B2C for authentication anyway.. conceivably could IdentityServer be removed from my abp application, and Azure AD B2C used in its place?

**Comment by rlavaud** [comment]
While it could and I have tried but ultimately I used IdentityServer to to connect to AD BC. The completely remove IdentityServer, you need to understand all the grants which needs to be setup and set them up in B2C, then you need to register your users after sign up and finally after a user logs in with B2C you have to connect to you app to load the the permissions in the claim (roles etc). In the end it is too much work. I found it simpler to set the users as external users so no password is stored then configure IdentityServer to use AD B2C as authentication source.

Remember Authentication != Authorization

**Comment by jhulbertpmn** [comment]
Thanks for the info. I've experimented with this a couple different ways and have had a similar experience. One concern which got me started on this process was having passwords stored in my database - as you said that can be addressed by having the users be external and having IdentityServer use AD B2C. The other concern is the coming license change related to IdentityServer - I'd prefer to not have to commercially license it if I don't have to, but perhaps the changes necessary to make that work are not worth the time investment. I'll continue experimenting with this on the side but  might just still with IdentityServer in the project for now. Thanks again!

**Comment by lommez** [comment]
@rlavaud, 

In my case, I am using the OpenIdDict template. Have you managed to implement IdentityServer using Azure B2C as an external authentication source? If so, would you have some code available demonstrating this or something to guide me in this implementation?
